tests/lib/rules/no-direct-mutation-state.js

```diff
@@ -70,6 +70,14 @@ ruleTester.run('no-direct-mutation-state', rule, {
       '}'
     ].join('\n')
   }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  constructor() {',
+      '    this.state.foo = 1;',
```

Comment:
Could we also add a test that makes `++` invalid inside the constructor?

Reply:
@TheSavior (cc: @zpao )

I actually have a PR proposing a rule catching all references to `this.state` used in at `this.setState` in #1374. Does this rule cover that as well? If not, I would be really happy for feedback.